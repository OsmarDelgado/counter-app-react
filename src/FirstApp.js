import React from "react"
import PropTypes from "prop-types"
// import React, { Fragment } from "react"

const FirstApp = ( { hello, subtitle } ) => {
    // const greetings = 'Hello World'
    // const object = {
    //     name: 'Osmar',
    //     age: 26
    // }

    return (
        <>
            <h1>{ hello }</h1>
            {/* <pre>{ JSON.stringify( object, null, 4 ) }</pre> */}
            <p>{ subtitle }</p>
        </>
    )
}

FirstApp.propTypes = {
    hello : PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle : 'Subtitle'
}

export default FirstApp
