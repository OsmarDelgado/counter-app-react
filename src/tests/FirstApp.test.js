import React from 'react'
import FirstApp from '../FirstApp'
import { shallow } from 'enzyme'

describe( 'Test in FirstApp', () => {
    // test( 'Shold show message "Hello, I am Osmar"', () => {
    //     const greetings = "Hello, I am Osmar"

    //     const { getByText } = render( <FirstApp hello={ greetings } /> )

    //     expect( getByText( greetings ) ).toBeInTheDocument()
    // } )

    test( 'Should show app successfully', () => {
        const greetings = 'Hello, I am Osmar'
        const wrapper = shallow( <FirstApp hello={ greetings } /> )

        expect( wrapper ).toMatchSnapshot()
    } )

    test( 'Should show subtitile sent by props', () => {
        const greetings = 'Hello, I am Osmar'
        const subtitle = 'Subtitle!!!'
        const wrapper = shallow( <FirstApp hello={ greetings } subtitle={ subtitle } /> )

        const textPha = wrapper.find( 'p' ).text()

        expect( textPha ).toBe( subtitle )
    } )
} )