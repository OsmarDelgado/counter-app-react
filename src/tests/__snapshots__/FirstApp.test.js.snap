// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp Should show app successfully 1`] = `
<Fragment>
  <h1>
    Hello, I am Osmar
  </h1>
  <p>
    Subtitle
  </p>
</Fragment>
`;
